File: backend/services/auth-service/src/main/java/com/eventzen/auth/exception/EventZenException.java
package com.eventzen.auth.exception;

import org.springframework.http.HttpStatus;

public class EventZenException extends RuntimeException {

    private final HttpStatus status;
    private final String error;
    private final String code;

    public EventZenException(HttpStatus status, String error, String code, String message) {
        super(message);
        this.status = status;
        this.error = error;
        this.code = code;
    }

    public HttpStatus getStatus() {
        return status;
    }

    public String getError() {
        return error;
    }

    public String getCode() {
        return code;
    }
}

File: backend/services/ticketing-service/src/EventZen.Ticketing.Api/Common/EventZenException.cs
namespace EventZen.Ticketing.Api.Common;

public sealed class EventZenException : Exception
{
    public int StatusCode { get; }
    public string Error { get; }
    public string Code { get; }
    public object? Details { get; }

    public EventZenException(int statusCode, string error, string code, string message, object? details = null) : base(message)
    {
        StatusCode = statusCode;
        Error = error;
        Code = code;
        Details = details;
    }
}
